src/Agent.Service/Windows/AgentService.cs

```diff
@@ -151,7 +151,7 @@ private Process CreateAgentListener()
             string exeLocation = Assembly.GetEntryAssembly().Location;
             string agentExeLocation = Path.Combine(Path.GetDirectoryName(exeLocation), "Agent.Listener.exe");
             Process newProcess = new Process();
-            newProcess.StartInfo = new ProcessStartInfo(agentExeLocation, "run");
+            newProcess.StartInfo = new ProcessStartInfo(agentExeLocation, "run --startuptype windowsservice");
```

Comment:
have you try test auto update with windows service?
you may run into binary mismatch between the service host and the agent binary.
ex.
1. let's say your pr will merge and ship in 2.120.0 agent.
2. i configure a 2.117.0 agent on my box as windows service.
3. when i upgrade my agent from 2.117.0 to 2.120.0, all binary will be replaced.
4. however, since the agentservice.exe (2.117.0) is still running by windows SCM, the new agentservice.exe (2.120.0) won't get pick up untill SCM restart the service.
5. this will cause problem, since the 2.120.0 agent.listener.exe will be launched by 2.117.0 agentservice.exe which will not send --startuptype.
6. I think you need to force the service host update by crashing it.